misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/InvocationArgProcessor.java

```diff
@@ -134,6 +158,136 @@
+                // the args which are subtypes of "java.lang.Object". Therefore all the primitive typed arguments
+                // must be converted into their wrapper implementations (i.e. 'int' -> 'Integer'), before passing
+                // into the method.
+                if (getParameterTypeName(params.get(i)).equals(BVariableType.ANY.getString())) {
+                    argValue = getValueAsObject(context, argValue);
+                }
+                argValues.put(argName, argValue);
+                remainingParams.remove(argName);
+            } else if (argType == ArgType.REST) {
+                if (namedArgsFound) {
+                    throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                            "rest args are not allowed after named args."));
+                }
+
+                String restParamName = null;
+                for (Map.Entry<String, ParameterNode> entry : remainingParams.entrySet()) {
+                    SyntaxKind parameterType = entry.getValue().kind();
+                    if (parameterType == SyntaxKind.REST_PARAM) {
+                        restParamName = entry.getKey();
+                        break;
+                    }
+                }
+                if (restParamName == null) {
+                    throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                            "undefined rest parameter."));
+                }
+                restArgsFound = true;
+                argValues.put(restParamName, arg.getValue().evaluate().getJdiValue());
+                remainingParams.remove(restParamName);
+            }
+        }
+
+        for (Map.Entry<String, ParameterNode> entry : remainingParams.entrySet()) {
+            String paramName = entry.getKey();
+            SyntaxKind parameterType = entry.getValue().kind();
+            if (parameterType == SyntaxKind.REQUIRED_PARAM) {
+                throw new EvaluationException(String.format(EvaluationExceptionKind.CUSTOM_ERROR.getString(),
+                        "missing required parameter '" + paramName + "'."));
+            } else if (parameterType == SyntaxKind.DEFAULTABLE_PARAM) {
+                argValues.put(paramName + DEFAULTABLE_PARAM_SUFFIX, VMUtils.make(context, 0).getJdiValue());
+            }
+        }
+
+        return argValues;
+    }
+
+    private static String getParameterName(ParameterNode parameterNode) {
+        switch (parameterNode.kind()) {
+            case REQUIRED_PARAM:
+                return ((RequiredParameterNode) parameterNode).paramName().get().text();
+            case DEFAULTABLE_PARAM:
+                return ((DefaultableParameterNode) parameterNode).paramName().get().text();
+            case REST_PARAM:
+                return ((RestParameterNode) parameterNode).paramName().get().text();
```

Comment:
Will `parameterNode).paramName()` always present? Shall we add `isPresent` check for `parameterNode).paramName()` ?

Reply:
+1, added in 8a8a248